deploy/cmd/event-handlers.go

```diff
@@ -0,0 +1,149 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/ovrclk/akash/provider/gateway"
+	"github.com/ovrclk/akash/provider/manifest"
+	"github.com/ovrclk/akash/pubsub"
+	dtypes "github.com/ovrclk/akash/x/deployment/types"
+	mtypes "github.com/ovrclk/akash/x/market/types"
+	pmodule "github.com/ovrclk/akash/x/provider"
+	ptypes "github.com/ovrclk/akash/x/provider/types"
+	"github.com/pkg/errors"
+)
+
+var (
+	errUnreachableCode = errors.New("should be unreachable code, exit with this error")
+)
+
+// EventHandler is a type of function that handles events coming out of the event bus
+type EventHandler func(pubsub.Event) error
+
+// SendManifestHander sends manifests on the lease created event
+func SendManifestHander(clientCtx client.Context, dd *DeploymentData) func(pubsub.Event) error {
+	return func(ev pubsub.Event) (err error) {
+		addr := clientCtx.GetFromAddress()
+		log := logger.With("action", "send-manifest")
+		switch event := ev.(type) {
+		// Handle Lease creation events
+		case mtypes.EventLeaseCreated:
+			if addr.String() == event.ID.Owner {
```

Comment:
It looks like this just checks for a matching owner. This could cause some issues if someone ran a bunch of copies of this tool at the same time to create multiple deployments.